jwst/assign_wcs/nircam.py

```diff
@@ -1,35 +1,49 @@
 from __future__ import (absolute_import, unicode_literals, division,
                         print_function)
+import logging
 
 from asdf import AsdfFile
+from copy import deepcopy
 from astropy import coordinates as coord
 from astropy import units as u
-from astropy.modeling.models import Scale
-
+from astropy.modeling.models import Scale, Identity
 import gwcs.coordinate_frames as cf
+from gwcs.wcs import WCS
+
 from . import pointing
-from .util import not_implemented_mode, subarray_transform
+from .util import not_implemented_mode, subarray_transform, get_object_info
+from ..datamodels import ImageModel, NIRCAMGrismModel
+from ..transforms.models import (GrismObject,
+                                 NIRCAMGrismModelSelector,
+                                 NIRCAMBackwardGrismDispersion)
 
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
 
-def create_pipeline(input_model, reference_files):
-    '''
-    get reference files from crds
 
-    '''
+def create_pipeline(input_model, reference_files, catalog=None):
+    """Get reference files from crds."""
     exp_type = input_model.meta.exposure.type.lower()
-    pipeline = exp_type2transform[exp_type](input_model, reference_files)
+    if catalog is not None:
+        pipeline = exp_type2transform[exp_type](input_model,
+                                                reference_files,
+                                                catalog)
+    else:
+        pipeline = exp_type2transform[exp_type](input_model, reference_files)
 
     return pipeline
 
 
 def imaging(input_model, reference_files):
-    """
-    The NIRCAM imaging pipeline includes 3 coordinate frames -
-    detector, focal plane and sky
+    """ The NIRCAM imaging pipeline
```

Comment:
apologies, I think that happened awhile ago and my editor marks through pep8/linter stuff. Has JWST decided on a particular  formatting for docstrings? I see them a few different ways in the code, but I'll be wary of messing with functions I know I'm not touching :-)  We should probably decide on a consistent way to go at some point since it makes a difference for the sphinx docs